src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -205,15 +211,40 @@ private string CreateMergedRuleset(AnalyzerSettings languageSpecificSettings)
                 return languageSpecificSettings.RuleSetFilePath;
             }
 
+            string resolvedRulesetPath = GetAbsoluteRulesetPath();
+
             var mergedRulesetFilePath = Path.Combine(ProjectSpecificOutputDirectory, "merged.ruleset");
             Log.LogMessage(MessageImportance.Low, Resources.AnalyzerSettings_CreatingMergedRuleset, mergedRulesetFilePath);
 
-            var content = GetMergedRuleSetContent(OriginalRulesetFilePath, languageSpecificSettings.RuleSetFilePath);
+            var content = GetMergedRuleSetContent(resolvedRulesetPath, languageSpecificSettings.RuleSetFilePath);
 
             File.WriteAllText(mergedRulesetFilePath, content);
             return mergedRulesetFilePath;
         }
 
+        private string GetAbsoluteRulesetPath()
+        {
+            // If the supplied ruleset path is relative then it is relative to the project folder.
+            // This relative path will be wrong if use it directly in the generated merged ruleset
+            // file so we need to make it absolute.
+            string resolvedRulesetFilePath;
+            if (Path.IsPathRooted(OriginalRulesetFilePath))
+            {
+                Log.LogMessage(MessageImportance.Low, $"Supplied ruleset path is rooted: {OriginalRulesetFilePath}");
+                resolvedRulesetFilePath = OriginalRulesetFilePath;
+            }
+            else
+            {
+                Log.LogMessage(MessageImportance.Low, $"Supplied ruleset path is not rooted: {OriginalRulesetFilePath}");
+                resolvedRulesetFilePath = Path.GetFullPath(Path.Combine(CurrentProjectDirectoryPath, OriginalRulesetFilePath));
+            }
+
+            Log.LogMessage(MessageImportance.Low,
+                File.Exists(resolvedRulesetFilePath) ? Resources.AnalyzerSettings_ResolvedRulesetFound : Resources.AnalyzerSettings_ResolvedRulesetNotFound,
+                resolvedRulesetFilePath);
```

Comment:
Specifying a non-existent ruleset doesn't cause a build failure, and I didn't notice any warning from MSBuild either so I've added some logging so we can at least diagnose what is happening.